### PR TITLE
test(melange): show dependency cycle depending on runtime_deps in target dir

### PR DESCRIPTION
**Pull request by anmonteiro**

### CODE REVIEW

File: test/blackbox-tests/test-cases/melange/emit-with-runtime-deps-edge-cases.t [hunk 1]
```diff
@@ -53,11 +53,9 @@ The runtime_dep index.txt was copied to the build folder
   hello from file
   
 
-
-
 Test depending on non-existing paths
 
-  $ mkdir another
+  $ mkdir -p another/another-output/another
```
Comment (anmonteiro) [inline test/blackbox-tests/test-cases/melange/emit-with-runtime-deps-edge-cases.t:58]: I had only ran the action builder eagerly because of this test, but it looks like creating its target dir in source makes the error pop up again, so it's rather an instance of #7207 and the recursive alias work that's gonna come after it.

File: test/blackbox-tests/test-cases/melange/emit-with-runtime-deps-dir-target.t [hunk 2]
```diff
@@ -31,7 +31,9 @@ Rules created for the assets in the output directory
   [1]
 
   $ dune build @mel --display=short
+          bash some_dir
```
Comment (rgrinberg) [inline test/blackbox-tests/test-cases/melange/emit-with-runtime-deps-dir-target.t:34]: In a separate PR, it would be good to switch this test to `sh` unless you need some special bash features.
